examples/rbac_with_domains_hierarchical_policy.csv

```diff
@@ -0,0 +1,6 @@
+p, admin, dom3, data, read
```

Comment:
Also little more complexity to the domain is warranted. At least one extra user to test out that they do not have access to domain1 if they only have access to domain2

Reply:
Added more tests